### DISCUSSION TITLE
how does drift relate to sqlite3 package?

### DISCUSSION THREAD

**Discussion by imoldfella** [simolus3/drift#1864] [General]
They seem closely related, but not obvious how they are connected. For example can you use Drift generated sql with CommonDatabase?

**Comment by imoldfella** [comment]
the drift documentation says to use sql-wasm.wasm, but when I look at the example todo app, it has sqlite3.wasm. So I think its fair to say that Drift is currently built on sqlite3.

Fwiw I needed to disable web worker flag to get the web todo app to run on the mac. _useWorker = false works, _useWorker = true is permanent beach ball.

**Reply by imoldfella** [reply]
So I figured out from the example app how to turn web and native sqlite3 into DatabaseConnection.
Can you reverse this: Can I get the raw-ish sqlite3 database from a DatabaseConnection or should I just keep both around.

I read that sqlite3 is fine from different isolates, but does that extend to web workers and the IndexedDbFileSystem? Different tabs?

**Reply by simolus3** [reply]
> Can I get the raw-ish sqlite3 database from a DatabaseConnection or should I just keep both around.

You should keep both around. It's true that the two "main" drift database implementations are based on the `sqlite3` package, but drift is general enough to support other implementations too. So we can't really expose the underlying sqlite3 database through that.

> Fwiw I needed to disable web worker flag to get the web todo app to run on the mac. _useWorker = false works, _useWorker = true is permanent beach ball.

Thanks, I'll take a look.

> I read that sqlite3 is fine from different isolates, but does that extend to web workers and the IndexedDbFileSystem? Different tabs?

A shared worker is shared between tabs, which is very close to how a `DriftIsolate` works too. You can even synchronize stream queries across tabs in real time.
So ideally the answer is to just use a (shared) web worker here, I'll take a look and see if I can reproduce the example not working with that setup.

**Reply by simolus3** [reply]
> Fwiw I needed to disable web worker flag to get the web todo app to run on the mac. _useWorker = false works, _useWorker = true is permanent beach ball.

Did you run `flutter pub run build_runner build` before running the app? That will generate the worker js file. It works for me after doing that. If that doesn't help in your case, could you open the console in Chrome and see if an interesting error pops up?

**Reply by imoldfella** [reply]
I did not! Works perfectly after that.
